internal/test/azuremachinepool/builder.go

```diff
@@ -0,0 +1,131 @@
+package azuremachinepool
+
+import (
+	"encoding/json"
+	"math/rand"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/apiextensions/v3/pkg/label"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	expcapzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+)
+
+type BuilderOption func(azureMachinePool *expcapzv1alpha3.AzureMachinePool) *expcapzv1alpha3.AzureMachinePool
+
+var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+func AcceleratedNetworking(acceleratedNetworking *bool) BuilderOption {
+	return func(azureMachinePool *expcapzv1alpha3.AzureMachinePool) *expcapzv1alpha3.AzureMachinePool {
+		azureMachinePool.Spec.Template.AcceleratedNetworking = acceleratedNetworking
+		return azureMachinePool
+	}
+}
+
+func Cluster(clusterName string) BuilderOption {
+	return func(azureMachinePool *expcapzv1alpha3.AzureMachinePool) *expcapzv1alpha3.AzureMachinePool {
+		azureMachinePool.Labels[capiv1alpha3.ClusterLabelName] = clusterName
+		return azureMachinePool
+	}
+}
+
+func DataDisks(dataDisks []capzv1alpha3.DataDisk) BuilderOption {
+	return func(azureMachinePool *expcapzv1alpha3.AzureMachinePool) *expcapzv1alpha3.AzureMachinePool {
+		azureMachinePool.Spec.Template.DataDisks = dataDisks
+		return azureMachinePool
+	}
+}
+
+func Location(location string) BuilderOption {
+	return func(azureMachinePool *expcapzv1alpha3.AzureMachinePool) *expcapzv1alpha3.AzureMachinePool {
+		azureMachinePool.Spec.Location = location
+		return azureMachinePool
+	}
+}
+
+func Name(name string) BuilderOption {
+	return func(azureMachinePool *expcapzv1alpha3.AzureMachinePool) *expcapzv1alpha3.AzureMachinePool {
+		azureMachinePool.ObjectMeta.Name = name
+		azureMachinePool.Labels[label.MachinePool] = name
+		return azureMachinePool
+	}
+}
+
+func StorageAccountType(storageAccountType compute.StorageAccountTypes) BuilderOption {
+	return func(azureMachinePool *expcapzv1alpha3.AzureMachinePool) *expcapzv1alpha3.AzureMachinePool {
+		azureMachinePool.Spec.Template.OSDisk.ManagedDisk.StorageAccountType = string(storageAccountType)
+		return azureMachinePool
+	}
+}
+
+func VMSize(vmsize string) BuilderOption {
+	return func(azureMachinePool *expcapzv1alpha3.AzureMachinePool) *expcapzv1alpha3.AzureMachinePool {
+		azureMachinePool.Spec.Template.VMSize = vmsize
+		return azureMachinePool
+	}
+}
+
+func BuildAzureMachinePool(opts ...BuilderOption) *expcapzv1alpha3.AzureMachinePool {
+	nodepoolName := generateName()
+	azureMachinePool := &expcapzv1alpha3.AzureMachinePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nodepoolName,
+			Namespace: "org-giantswarm",
+			Labels: map[string]string{
+				label.AzureOperatorVersion: "5.0.0",
+				label.Cluster:              "ab123",
```

Comment:
Maybe add `capiv1alpha3.ClusterLabelName` here and below (`MachinePool`) as well?